### PR TITLE
add pull-test-infra-bazel-canary for testing the latest bazelbuild

### PR DESCRIPTION
**Pull request by BenTheElder**
This should help when working to upgrade the bazelbuild image.

### CODE REVIEW

**Comment by BenTheElder** [comment]
/area bazel
/area jobs
cc @ixdy FYI, we can test this over at https://github.com/kubernetes/test-infra/pull/5137

File: prow/config.yaml [hunk 1]
```diff
@@ -2629,6 +2629,47 @@ presubmits:
...
+    agent: kubernetes
+    context: pull-test-infra-bazel-canary
+    branches:
+    - master
+    always_run: false
+    rerun_command: "/test pull-test-infra-bazel-canary"
+    trigger: "(?m)^/test pull-test-infra-bazel-canary,?(\\s+|$)"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/bazelbuild:latest-0.7.0
```
Comment (ixdy) [inline prow/config.yaml]: do we need to also set `imagePullPolicy: Always`?

Comment (BenTheElder) [reply prow/config.yaml]: done.

**Comment by ixdy** [comment]
we require more testgrid tabs

**Pull request merged**